src/main/java/com/globallabs/pots/TelephoneDevice.java

```diff
@@ -28,6 +26,32 @@ public void dial(final int phoneNumber) throws DialingMySelfException {
 			telephone.setStatus(Status.OFF_CALL);
 		}
 	}
+
+	public void ring() throws BusyPhoneException {
+		if (telephone.getStatus().equals(Status.DIALING)){
+			telephone.setStatus(Status.RINGING);
+			long start = System.currentTimeMillis();
+			long end = start + 10*1000;
+			while (System.currentTimeMillis() < end) {
```

Comment:
These lineas need a little more of explanation of what is happening @angelicacosta